src/logger.c

```diff
@@ -0,0 +1,127 @@
+#include "logger.h"
+
+#include "constants.h"
+
+#include <math.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+LoggerData g_logger;
+
+#ifndef min
```

Comment:
Only on the embedded platform.  The ordering causes a bit of trouble here, since it needs to be after all the includes. If I find myself using it anywhere else I'll move it into a header.